python/common/org/python/types/Str.java

```diff
@@ -543,10 +543,66 @@ else if (other instanceof org.python.types.Bool) {
     }
 
     @org.python.Method(
-        __doc__ = ""
-    )
-    public org.python.Object center() {
-        throw new org.python.exceptions.NotImplementedError("center() has not been implemented.");
+        __doc__ = "",
+		args = {"width"},
+		default_args = {"charToFill"}
+    )
+    public org.python.Object center(org.python.Object width, org.python.Object charToFill) {
+		if(this instanceof org.python.types.Str)
+		{
+			java.lang.String fillChar = new java.lang.String();
+			if(charToFill instanceof org.python.types.Str)
+			{
+				// fillChar has right type
+				if( ((org.python.types.Str)charToFill).value.length() != 1) {
+					throw new org.python.exceptions.TypeError("The fill character must be exactly one character long");
+				}
+				fillChar = ((org.python.types.Str) charToFill).value;
+			}
+			else if(charToFill == null)
+			{
+				fillChar = " ";
+			}
+			else {
+				throw new org.python.exceptions.TypeError("Fill char must of type String");
+			}
+			
+			if(width instanceof org.python.types.Int)
+			{
+				String str = this.value;
+				int widthVal = (int)((org.python.types.Int)width).value;
+				int strLen = str.length();
+				if(strLen >= widthVal )
+				{
+					return new org.python.types.Str(str);
+				}
+				else {
+					int diff = widthVal - strLen;
+					int lenFirst = (diff)/2;
+					int lenSecond = diff - lenFirst;
+					
+					java.lang.StringBuffer returnString = new java.lang.StringBuffer(widthVal);
+					
+					for(int i=0; i<lenFirst; i++) {
+						returnString.append(fillChar);
+					}
+					returnString.append(str);
+					for(int i=0; i<lenSecond; i++) {
```

Comment:
Similar style  issue here with the for; a space after `for`, and spaces around the `=` and `<` operators.